packages/react-ui-kit/src/Form/Select.tsx

```diff
@@ -134,39 +165,41 @@ const dropdownOptionStyles = (theme: Theme, isSelected: boolean): CSSObject => (
   '&:not(:first-of-type)': {
     borderTop: `1px solid ${theme.Select.borderColor}`,
   },
-  '&:hover, &:active, &:focus': {
-    background: theme.general.primaryColor,
-    borderColor: theme.general.primaryColor,
-    color: theme.Select.contrastTextColor,
-  },
 });
 
 const filterSelectProps = props => filterProps(props, ['markInvalid']);
 
 const placeholderText = '- Please select -';
 
-export const Select = <T extends SelectOption = SelectOption>({
-  id,
-  label,
-  error,
-  helperText,
-  options = [],
-  value = null,
-  onChange,
-  required,
-  markInvalid,
-  dataUieName,
-  wrapperCSS = {},
-  ...props
-}: SelectProps<T>) => {
-  const currentOptionIdx = options.findIndex(option => option.value === value?.value);
-  const selectedOption = currentOptionIdx === -1 ? null : currentOptionIdx;
+const isSingle = (props: SingleSelectProps | MultipleSelectProps): props is SingleSelectProps =>
+  !(props as SingleSelectProps).isMultiSelect;
+
+const isMultiple = (props: SingleSelectProps | MultipleSelectProps): props is MultipleSelectProps =>
+  (props as MultipleSelectProps).isMultiSelect;
+
+export const Select = <T extends SelectOption = SelectOption>(props: SingleSelectProps<T> | MultipleSelectProps<T>) => {
+  const {
+    id,
+    label,
+    error,
+    helperText,
+    options = [],
+    isMultiSelect = false,
+    value = null,
+    onChange,
+    required,
+    markInvalid,
+    dataUieName,
+    wrapperCSS = {},
+    ...rest
+  } = props;
 
   const selectContainerRef = useRef<HTMLDivElement>(null);
   const listRef = useRef<HTMLUListElement>(null);
+
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [selectedOption, setSelectedOption] = useState(null);
```

Comment:
seems like we may run into similar issues with internal state

Reply:
Yes, do we really need, to store selected option inside the component? This is basically what I've removed on last Friday. We should rather render select's UI based on its props (`value` props). 🤔